src/libraries/System.IO.Compression.Brotli/tests/CompressionStreamUnitTests.Brotli.cs

```diff
@@ -66,6 +68,105 @@ public void GetMaxCompressedSize_Basic()
+            {
+                var bs = new BrotliStream(ms, options);
+                foreach (byte b in correctUncompressedBytes)
+                {
+                    bs.WriteByte(b);
+                }
+                bs.Dispose();
+                compressedBytes = ms.ToArray();
+            }
+
+            byte[] decompressedBytes = new byte[correctUncompressedBytes.Length];
+            using (var ms = new MemoryStream(compressedBytes))
+            using (var bs = new BrotliStream(ms, CompressionMode.Decompress))
+            {
+                for (int i = 0; i < decompressedBytes.Length; i++)
+                {
+                    int b = bs.ReadByte();
+                    Assert.InRange(b, 0, 255);
+                    decompressedBytes[i] = (byte)b;
+                }
+                Assert.Equal(-1, bs.ReadByte());
+                Assert.Equal(-1, bs.ReadByte());
+            }
+
+            Assert.Equal<byte>(correctUncompressedBytes, decompressedBytes);
+        }
+
+        [Theory]
+        [MemberData(nameof(UncompressedTestFiles))]
+        public async void BrotliCompressionQuality_SizeInOrder(string testFile)
+        {
+            using var uncompressedStream = await LocalMemoryStream.readAppFileAsync(testFile);
+
+            async Task<long> GetLengthAsync(int compressionQuality)
+            {
+                uncompressedStream.Position = 0;
+                using var mms = new MemoryStream();
+                using var compressor = new BrotliStream(mms, new BrotliCompressionOptions() { Quality = compressionQuality });
+                await uncompressedStream.CopyToAsync(compressor);
+                await compressor.FlushAsync();
+                await uncompressedStream.FlushAsync();
+                return mms.Length;
+            }
+
+            long quality0 = await GetLengthAsync(0);
+            long quality1 = await GetLengthAsync(1);
+            long quality2 = await GetLengthAsync(2);
+            long quality3 = await GetLengthAsync(3);
+            long quality4 = await GetLengthAsync(4);
+            long quality5 = await GetLengthAsync(5);
+            long quality6 = await GetLengthAsync(6);
+            long quality7 = await GetLengthAsync(7);
+            long quality8 = await GetLengthAsync(8);
+            long quality9 = await GetLengthAsync(9);
+            long quality10 = await GetLengthAsync(10);
+            long quality11 = await GetLengthAsync(11);
+
+            Assert.True(quality1 <= quality0);
+            Assert.True(quality2 <= quality1);
+            Assert.True(quality3 <= quality1);
+            Assert.True(quality4 <= quality2);
+            Assert.True(quality5 <= quality1);
+            Assert.True(quality6 <= quality1);
+            Assert.True(quality7 <= quality4);
+            Assert.True(quality8 <= quality7);
+            Assert.True(quality9 <= quality8);
+            Assert.True(quality10 <= quality9);
+            Assert.True(quality11 <= quality10);
```

Comment:
I don't understand these comparisons. Why for example are 3, 5, and 6 being compared against 1?

Reply:
Most cases these were linear, but for some file types these were level2 is smaller than level3, i.e. the compressed size were not depending on compression level linearly, it depends on compression file type and strategy

Reply:
Ended up to only test with the files that produces linear 1 to 9 compression ratio